### DISCUSSION TITLE
Feature request: publish quantization-sensitive tensor guidance and reference validation for SenseNova-U1.5 GGUF

### DISCUSSION THREAD

**Discussion by GuXinghai** [OpenSenseNova/SenseNova-U1#274] [Ideas]
Thank you for releasing SenseNova-U1.5 and for continuing to improve GGUF / low-VRAM inference support.

The current README notes that **SenseNova-U1.5-Lite GGUF weights are coming soon**, and the official inference code already provides a `--gguf_checkpoint` path. I would like to suggest publishing a small amount of **quantization-sensitivity guidance** together with the official GGUF release.

The motivation is that model developers may be able to derive or publish information that downstream community quantizers generally cannot recover reliably from the final checkpoint alone.

Community quantizers usually have access only to the released weights, architecture, inference code, and limited end-to-end testing. As a result, tensor-precision decisions are often based on static weight metrics, architectural heuristics, previous GGUF experience, and trial-and-error.

A recent SenseNova-U1.5 Final Q4_K_S case illustrates this limitation:

[https://huggingface.co/realrebelai/SenseNova-U1.5-8B_GGUFs/discussions/3](https://huggingface.co/realrebelai/SenseNova-U1.5-8B_GGUFs/discussions/3)

A repaired artifact and the associated audit reports are available here:

[https://huggingface.co/GuYueshou/SenseNova-U1.5-8B-MoT-Q4_K_S-L0DownFix-GGUF](https://huggingface.co/GuYueshou/SenseNova-U1.5-8B-MoT-Q4_K_S-L0DownFix-GGUF)

In that specific artifact, the failure was localized to:

```text
language_model.model.layers.0.mlp.down_proj.weight
```

Under the tested 50-step SenseNova-U1.5 Final baseline, restoring only this tensor to the official Final BF16 values prevented the measured early-prefix divergence and restored normal semantic and structural generation.

The interesting part was that the problematic Q4_K tensor was actually closer to the official Final tensor than a behaviorally working Q4_0 control under global weight-space relative L2 error.

However, on measured real activation inputs, its activation-weighted operator error was worse.

The observed propagation path was:

```text
layer-0 understanding mlp.down_proj quantization error
-> early prefix hidden-state drift
-> severe prefix V-cache corruption
-> generation-attention failure
-> broken image trajectory
```

This suggests that:

> Global weight-space accuracy does not necessarily predict activation-weighted operator accuracy.

This result is intentionally narrow. It does **not** imply that Q4_K is generally inferior, that every SenseNova-U1.5 quantization has the same sensitive tensor, or that a single-tensor repair is sufficient for every prompt, mode, or quantization implementation.

There is some related prior community evidence as well. Issue #92 reported visible artifacts in an earlier GGUF quantization and speculated that `fm_modules` or output-layer precision might be involved:

The newer U1.5 Final case adds a more detailed causal localization rather than only an output-level observation.

This finding concerns the released SenseNova-U1.5 Final checkpoint. It is presented as evidence for the need for model-specific validation guidance, not as a claim that the forthcoming U1.5-Lite weights necessarily share the same tensor-sensitivity profile.

The official deployment documentation already treats understanding and generation as separately tunable precision domains. That makes it especially useful to expose a small amount of model-specific sensitivity guidance to downstream quantizers.

### Minimum useful deliverable

A simple manifest identifying modules or tensors that should not be aggressively quantized, for example:

```text
critical tensors/modules  -> keep BF16/F16
sensitive tensors/modules -> recommended minimum precision
ordinary projections      -> lower-bit quantization acceptable
```

Even a small list of known-sensitive modules would substantially reduce trial-and-error for community GGUF authors.

### Recommended

A small set of official calibration prompts and a reference quantized-checkpoint regression script covering representative paths such as:

* visual understanding;
* text-to-image generation;
* image editing;
* interleaved multimodal inference.

This would give community quantizers a common validation target rather than relying on a few subjective output samples.

### Optional

If practical, an additional coarse sensitivity classification such as:

```text
High / Medium / Low
```

or non-sensitive derived importance / activation statistics suitable for mixed-precision quantization.

This would not require publishing training data, gradients, proprietary training logs, or any internal training artifacts.

The goal is not to ask the official team to maintain every possible Q3/Q4/Q5/Q6 checkpoint.

Instead, the official project could provide the model-specific quantization prior, while the community continues to produce hardware- and memory-targeted GGUF variants.

This would make mixed-precision quantization much less dependent on reverse-engineering tensor sensitivity from the final checkpoint alone, and could significantly improve the reliability of community low-VRAM releases.

Thank you for considering this suggestion.
